### PR TITLE
Fix mapping costs

### PR DESCRIPTION
**Pull request by jesper-friis**
Corrected calculation of mapping costs.

TODO: check MappingStep.eval() including MappingStep.get_inputs()

### CODE REVIEW

File: bindings/python/mappings.py [hunk 1]
```diff
@@ -228,14 +229,25 @@ def lowest_costs(self, nresults=5):
...
         n = 0
         for inputs in self.input_routes:
-            owncost = 1
-            for cost, idx in get_lowest_costs(inputs, nresults=nresults):
-                if isinstance(self.cost, Callable):
-                    values = get_values(inputs, idx, magnitudes=True)
-                    owncost = self.cost(**values)
+            costs = np.array([0.0])
+            for input in inputs.values():
+                if isinstance(input, MappingStep):
```
Comment (francescalb) [inline bindings/python/mappings.py:234]: Can you add a comment on how costs are calculated for MappingStep?

File: bindings/python/mappings.py [hunk 2]
```diff
@@ -228,14 +229,25 @@ def lowest_costs(self, nresults=5):
...
+                    costs = np.tile(costs, len(res)) + np.repeat(
+                        [c for c, _ in sorted(res, key=lambda x: x[1])],
+                        len(costs))
                 else:
-                    owncost = self.cost
-                result.append((cost + owncost, n + idx))
+                    costs += input.cost
+
+            res = sorted((c, i) for i, c in enumerate(costs))[:nresults]
+            if isinstance(self.cost, Callable):
```
Comment (francescalb) [inline bindings/python/mappings.py:243]: Can you add a comment on how costs are calculated for Callable and how this differs from MappingStep?

File: bindings/python/mappings.py (hunk 1 above)
Comment (jesper-friis) [reply bindings/python/mappings.py:250]: Done

File: bindings/python/mappings.py (hunk 2 above)
Comment (jesper-friis) [reply bindings/python/mappings.py:264]: Done

**Review by francescalb — APPROVED** [review]
Approved under condition that an issue is made about a more thorough description of the algorithm and thoughts behind :)

**Pull request merged**